core/src/main/java/org/elasticsearch/search/Scroll.java

```diff
@@ -38,18 +38,28 @@
     private final TimeValue keepAlive;
 
     public Scroll(StreamInput in) throws IOException {
-        this.keepAlive = new TimeValue(in);
-    }
+        TimeValue keepAlive = new TimeValue(in);
+        Objects.requireNonNull(keepAlive,"keepAlive must not be null");
+        if(keepAlive.seconds() > 300){
```

Comment:
Consider creating a constant for the timeout value instead of using a magic number.
Perhaps something like `public static final TimeValue TIMEOUT = timeValueSeconds(300)` if you want to be able to convert between time units. 
Otherwise `public static final int TIMEOUT_SECONDS = 300` should be sufficient.